scripts/horton-regression-test.py

```diff
@@ -0,0 +1,318 @@
+def capture_io():
+    """Context manager to capture standard output and error temporarily."""
+    old_stdout = sys.stdout
+    old_stderr = sys.stderr
+    sys.stdout = StringIO.StringIO()
+    sys.stderr = StringIO.StringIO()
+    try:
+        yield sys.stdout, sys.stderr
+    finally:
+        sys.stdout = old_stdout
+        sys.stderr = old_stderr
+
+
+def main():
+    """Main program."""
+    # Get command-line args
+    example_paths, do_update, do_force, do_capture = parse_args()
+
+    # Exit status becomes 1 when some test results have changed beyond the threshold, or
+    # when previous or reference values are missing.
+    exit_status = 0
+
+    # Run all examples.
+    for example_path in example_paths:
+        print example_path
+
+        # Sanity checks
+        if not example_path.endswith(".py"):
+            raise RegressionTestError('Example path should end with ".py": {}.'
+                                      .format(example_path))
+        with open(example_path) as f:
+            example_lines = f.readlines()
+            try:
+                example_lines.index(comment_regtest)
+            except ValueError:
+                raise RegressionTestError('Missing regression-testing comment in example {}.'
+                                          .format(example_path))
+
+        # Run example and collect results.
+        example_globals = {}
+        with open(example_path) as fh:
+            if do_capture:
+                with capture_io():
+                    exec fh in example_globals  # pylint: disable=exec-used
+            else:
+                exec fh in example_globals  # pylint: disable=exec-used
+
+        # Check that rt_results is defined in the example.
+        if 'rt_results' not in example_globals:
+            raise RegressionTestError('No rt_results defined in example {}.'
+                                      .format(example_path))
+
+        # Extract relevant variables from example.
+        rt_results = example_globals['rt_results']
+        rt_thresholds = example_globals.get('rt_thresholds', {})
+        rt_references = example_globals.get('rt_references', {})
+        rt_previous = example_globals.get('rt_previous', {})
+        known_keys = ['rt_results', 'rt_thresholds', 'rt_references', 'rt_previous']
+        for key in example_globals:
+            if key.startswith('rt_') and key not in known_keys:
+                print 'Unrecognized variable starting with rt_: {}.' % key
+
+        # Compare results with expected or previous values, if previous values are present.
+        rt_status = {}
+        for key, result in rt_results.iteritems():
+            target = rt_references.get(key, rt_previous.get(key))
+            if target is None:
+                rt_status[key] = 'PREVIOUS OR REFERENCE MISSING'
+            else:
+                threshold = rt_thresholds.get(key, default_threshold)
+                if np.allclose(result, target, atol=threshold, rtol=1e100):
```

Comment:
Oops. I just checked the documentation and this is very wrong. It should be zero instead.